pkg/kubelet/kubelet_getters.go

```diff
@@ -109,6 +109,22 @@ func (kl *Kubelet) ListPodsFromDisk() ([]types.UID, error) {
 	return kl.listPodsFromDisk()
 }
 
+// HandlerSupportsUserNamespaces checks whether the specified handler supports
+// user namespaces.
+func (kl *Kubelet) HandlerSupportsUserNamespaces(runtimeHandler string) (bool, error) {
+	kl.updateRuntimeMux.Lock()
+	defer kl.updateRuntimeMux.Unlock()
```

Comment:
The doc for updateRuntimeMux says:
```
         // updateRuntimeMux is a lock on updating runtime, because this path is not thread-safe.
        // This lock is used by Kubelet.updateRuntimeUp and Kubelet.fastNodeStatusUpdate functions and shouldn't be used anywhere else.
```

Shall the doc be updated, maybe?

I don't think we can avoid using that mutex, the `kl.containerRuntime.Status()` is called there, so we need to sync between the caller that stores what is read from the runtime and the getter that reads it (`HandlerSupportsUserNamespaces()`).

Reply:
This is fixed too